### ISSUE TITLE
 what is the performance training from scratch 

### ISSUE THREAD

**Issue by zkk0911** [wizyoung/YOLOv3_TensorFlow#11]
Thanks for your meticulous works. Have you trained from scratch using your codes here and what is the performance？ I find there are  many differences with original keras codes

**Comment by wizyoung** [comment]
On my own project dataset, I trained this YOLO v3 from scratch and the performance was way better than Faster-RCNN and SSD (I was using the official tf object detection api). By the way, I used many skills like data augmentation mentioned [here](https://github.com/wizyoung/YOLOv3_TensorFlow#9-other-tricks).

**Comment by wizyoung** [comment]
By the way, there is no official keras code. My version is consistent with the original setting of the official paper and code. I've seen many versions like Gluon-CV changing the loss function and training strategy for better performance.

**Comment by OPPOA113** [comment]
@wizyoung 你好，你那个数据增强部分代码能加到仓库吗，--苦笑--。
还有，多GPU的训练在哪设置呢？

**Comment by wizyoung** [comment]
@OPPOA113 数据增强部分代码比较乱，还未整理，你可以暂时参考 [link](https://blog.paperspace.com/data-augmentation-for-object-detection-rotation-and-shearing/)。
多 GPU 的代码时间原因还没能抽时间完善，说一下基本思路：模型放 CPU 上，多个 GPU 更新参数，可以参考我这份[代码](https://github.com/wizyoung/I3D_TensorFlow/blob/master/finetune_multi_gpu.py)。

最近比较忙，可能要等一段时间更新。欢迎 PR。

**Comment by zkk0911** [comment]
我使用了warm-up策略（设置use_warm_up=True），然后restore_part=None(restore all parameters)、  update_part=None(update the whole model)在COCO数据集上从头开始训练，但是第5（warm_up_epoch）个之后，又出现了loss为nan的情况， 哪里有问题呢

**Comment by wizyoung** [comment]
@zkk0911 参考 [#3](https://github.com/wizyoung/YOLOv3_TensorFlow/issues/3#issuecomment-456818802). 这个问题并没有彻底解决的方法。

**Comment by wizyoung** [comment]
@zkk0911 @OPPOA113 已有朋友自行完善了数据增强和其他相关训练策略，告知我在 VOC 上达到了 86% 的 mAP。请自行完善。

**Comment by Stinky-Tofu** [comment]
@zkk0911 @OPPOA113 If you want to get better performance, you can reference my code https://github.com/Stinky-Tofu/Stronger-yolo

**Comment by wizyoung** [comment]
@Stinky-Tofu That's really a great job!

**Comment by Stinky-Tofu** [comment]
@wizyoung Thank you!

**Comment by zkk0911** [comment]
@wizyoung Thanks
